### PR TITLE
GCE support.

### PR DESCRIPTION
**Pull request by enxebre**

### CODE REVIEW

File: bootstrap/aws-public/config-default.sh [hunk 1]
```diff
@@ -19,6 +19,8 @@ export TF_VAR_master_size=${TF_VAR_master_size:-m1.medium}
 export TF_VAR_slave_size=${TF_VAR_slave_size:-m1.medium}
 export TF_VAR_slaves=${TF_VAR_slaves:-1}
 export TF_VAR_subnet_availability_zone=${TF_VAR_subnet_availability_zone:-eu-west-1a}
+# Overrides default folder in Terraform.py inventory.
+export TFSTATE_ROOT=`pwd`/terraform/aws-public
```
Comment (tayzlor) [inline bootstrap/aws-public/config-default.sh]: this cant be the pwd of this script? is it better to make it relative to $APOLLO_ROOT?

File: terraform/gce/mesos-slave.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  type    = "googlecompute.image"
+  version = "${var.atlas_artifact_version.slave}"
+}
+
+resource "google_compute_instance" "mesos-slave" {
+    count     = "${var.slaves}"
+    name         = "apollo-mesos-slave-${count.index}"
+    machine_type = "${var.instance_type.slave}"
+    zone = "${var.zone}"
+    tags = ["mesos-slave","http","https","ssh"]
```
Comment (tayzlor) [inline terraform/gce/mesos-slave.tf]: tidy up formatting here

File: terraform/gce/variables.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
+variable "account_file" {
```
Comment (tayzlor) [inline terraform/gce/variables.tf]: tidy up formatting in this file

File: packer/ubuntu-14.04_amd64-google.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
   },
   "builders": [{
     "type": "googlecompute",
-    "image_name": "apollo-ubuntu-14.04-amd64",
+    "image_name": "apollo-ubuntu-1404-amd64",
```
Comment (tayzlor) [inline packer/ubuntu-14.04_amd64-google.json]: do GCE not accept . in their image names?

**Comment by tayzlor** [comment]
looks quite good to start with. needs a thorough testing across providers.

File: bootstrap/aws-public/config-default.sh (hunk 1 above)
Comment (enxebre) [inline bootstrap/aws-public/config-default.sh]: Yeah, in theory this is so being loade when running "launch" so from project root, was planing to improve it with refactoring in #285

File: packer/ubuntu-14.04_amd64-google.json (hunk 4 above)
Comment (enxebre) [inline packer/ubuntu-14.04_amd64-google.json]: Apparently no

**Comment by tayzlor** [comment]
Still TODO on this - 

- [ ] Fork https://github.com/CiscoCloud/terraform.py to Capgemini and do pull request for Digitalocean inventory
- [ ] Pull inventory file/repo into here via PIP
- [ ] Amend TF scripts for DO

**Comment by enxebre** [comment]
I am just using curl for retrieving the file.

File: bootstrap/gce/util.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  if [[ "$(which ansible-playbook)" == "" ]]; then
+    echo -e "${color_red}Can't find ansible-playbook in PATH, please fix and retry.${color_norm}"
+    exit 1
+  fi
+  if [[ "$(which python)" == "" ]]; then
+    echo -e "${color_red}Can't find python in PATH, please fix and retry.${color_norm}"
+    exit 1
+  fi
+  if [[ "$(pip list | grep boto)" == "" ]]; then
```
Comment (tayzlor) [inline bootstrap/gce/util.sh]: does gce use boto ?

Comment (enxebre) [inline bootstrap/gce/util.sh]: Actually any provider need it now.
Those scripts would be removed/refactor with https://github.com/Capgemini/Apollo/pull/290

**Pull request merged**